### PR TITLE
Ch code cleanup

### PR DESCRIPTION
**Pull request by kasambuli**
Refactor code to make it more DRY

### CODE REVIEW

File: src/components/Form.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        border: "1px solid rgba(0, 0, 0, 0.5)",
+        boxSizing: "border- box",
+        lineHeight: "28px",
+    }
+}))
+
+const Form = ({ headingText, inputLength, inputType, variant, placeholder }) => {
+    const classes = useStyles();
+    const renderInput = () => {
+        if (inputLength == "2") {
```
Comment (HenryKenya) [inline src/components/Form.js]: why is this here?

Comment (HenryKenya) [reply src/components/Form.js]: for some reason, I think the LOC for the function renderInput can be reduced

File: src/data/constants.js [hunk 2]
```diff
@@ -103,34 +85,19 @@ export const AUTHENTICATION_BUTTONS = [
...
-	},
-	{
-		text: "Activity Here"
-	},
-	{
-		text: "Activity Here"
-	},
-];
+export const REGISTRATION_INPUTS = [
+	{ placeholder: "Username", type: "text" },
```
Comment (HenryKenya) [inline src/data/constants.js]: these could have been supplied as props instead of constants

File: src/views/Authentication/Registration.js [hunk 3]
```diff
@@ -1,48 +1,17 @@
...
-							<Input type="text" placeholder="Email Address" className={classes.inputs} />
-							<Input type="text" placeholder="Create Password" className={classes.inputs} />
-							<Input type="text" placeholder="Confirm Password" className={classes.inputs} />
-						</div>
-						<div className={classes.buttonDiv} >
-							<Button text="Register" />
-							<p> <a href="#">Sign In</a></p>
-						</div>
-					</form>
+					<Form headingText="Register" variant="outlined" inputLength="4" />
```
Comment (HenryKenya) [inline src/views/Authentication/Registration.js]: highly coupled `Form` with the props input length. Consider passing refactoring to avoid this

**Review by HenryKenya — CHANGES_REQUESTED** [review]
Left a couple of comments on areas to refactor better

File: src/components/Form.js (hunk 1 above)
Comment (HenryKenya) [reply src/components/Form.js]:
```
const Form = ({text, inputProps}) => {
  const renderInput = () => {
     inputProps.map( prop => ( <Input inputType={prop.inputType} placeholder={prop.placeholder} variant={prop.variant} className={prop.className }/>))
  }
// rest of the code
}
```

File: src/views/Authentication/Registration.js (hunk 3 above)
Comment (HenryKenya) [reply src/views/Authentication/Registration.js]: then supply inputProps as an array

File: src/data/constants.js (hunk 2 above)
Comment (HenryKenya) [reply src/data/constants.js]: move this to the inputProps of the individual component

**Comment by kasambuli** [comment]
> Left a couple of comments on areas to refactor better

Pushed changes to the branch
